### PR TITLE
Docker Image documentation; updated and transferred from separate repository Readme

### PR DESCRIPTION
**Pull request by rdebusscher**
1: Transfer of the readme files of the archived Github repo containing Docker Image usage.

This is the basis to start improved and updated Payara Docker Image usage.

### CODE REVIEW

**Comment by rdebusscher** [comment]
Updated the documentation to current state of Docker Image.

**Comment by OndroMih** [comment]
Should we add info about where to find Payara Enterprise Docker Images too? Or just in the Enterprise docs?

**Comment by OndroMih** [comment]
The page `docker-images.adoc` should be modified too. The following sentence is out of date:

```
Payara Services Limited provides Docker Images for the Payara Platform and updates them with every public release.
```
It should be more like:
```
Payara Services Limited provides Docker Images for the Payara Platform Community and updates them with every release.
```

And at the bottom of the page we could mention that Payara Platform Enterprise Docker Images are available in a private repository for Payara Enterprise customers.

File: docs/modules/ROOT/pages/documentation/ecosystem/docker-server-usage.adoc [hunk 1]
```diff
@@ -0,0 +1,174 @@
...
+If you do not want to create a sub-image, you can also mount a volume to `/opt/payara/scripts/init.d` and each `*.sh` file in there will be executed in standard file order.
+
+*Please note:* you can combine both approaches, but please keep in mind that scripts from `init.d` will run _after_ those from subimages!
+
+=== *The default Docker entry point*
+
+The default entry point is https://github.com/krallin/tini[tini], as the JVM should not run as PID 1. The default `CMD` argument for `tini` runs the `bin/entrypoint.sh` script in _exec_ mode, which in turn runs the following:
+
+* `${SCRIPT_DIR}/init_1_generate_deploy_commands.sh`. This script outputs deploy commands to the post boot command file located at `$POSTBOOT_COMMANDS` (default `$CONFIG_DIR/post-boot-commands.asadmin`). If the deploy commands are already found in that file, this script does nothing.
+* `${SCRIPT_DIR}/init_*.sh` scripts that you may provide for custom use as waiting or initializing during startup, *before* Glassfish kicks in.
```
Comment (OndroMih) [inline docs/modules/ROOT/pages/documentation/ecosystem/docker-server-usage.adoc:158]:
```suggestion
* `${SCRIPT_DIR}/init_*.sh` scripts that you may provide for custom use as waiting or initializing during startup, *before* Payara Server starts.
```

File: docs/modules/ROOT/pages/documentation/ecosystem/docker-micro-usage.adoc [hunk 2]
```diff
@@ -0,0 +1,165 @@
...
+ --deployFromGAV "fish.payara.examples:my-application:1.0-SNAPSHOT" \
+ --additionalRepository https://172.17.0.10/content/repositories/snapshots
+....
+
+==== *Define context root*
+
+_Since Payara Micro 5.191_
+
+
+The context root of the deployed application(s) is determined based on the file name of the artifact which is deployed. Form Payara Micro version 5.191 on, you can explicitly define the value of the context root using the `--contextroot` option, including deploying the application on the root itself ( `/`, specify *ROOT* as option value).
```
Comment (OndroMih) [inline docs/modules/ROOT/pages/documentation/ecosystem/docker-micro-usage.adoc:103]: "From version" not needed, it's already specified above.

```suggestion
The context root of the deployed application(s) is determined based on the file name of the artifact which is deployed. You can explicitly define the value of the context root using the `--contextroot` option, including deploying the application on the root itself ( `/`, specify *ROOT* as option value).
```

File: docs/modules/ROOT/pages/documentation/ecosystem/docker-micro-usage.adoc [hunk 3]
```diff
@@ -0,0 +1,165 @@
...
+COPY myapplication.war $DEPLOY_DIR
+
+CMD ["--deploymentDir", "/opt/payara/deployments", "--contextroot", "my"]
+....
+
+Creates a new Docker Image which results in the deployment of the _myapplication.war_ under the root context _/my_.
+
+== Speeding up Boot by disabling the Data Grid
+
+By default the image has the Data Grid enabled and instances of Payara Micro will form a Data Grid if they can see each other over multicast. Typically this will occur on a single host.
```
Comment (OndroMih) [inline docs/modules/ROOT/pages/documentation/ecosystem/docker-micro-usage.adoc:138]: The last sentence doesn't make any sense to me.

```suggestion
By default the image has the Data Grid enabled and instances of Payara Micro will form a Data Grid if they can see each other over multicast. 
```

File: docs/modules/ROOT/pages/documentation/ecosystem/docker-micro-usage.adoc [hunk 4]
```diff
@@ -0,0 +1,165 @@
...
+
+To speed up boot time the Data Grid can be disabled by modifying the CMD to add `--nocluster`
```
Comment (OndroMih) [inline docs/modules/ROOT/pages/documentation/ecosystem/docker-micro-usage.adoc:140]:
```suggestion
If you don't intend to form a cluster and want to speed up boot time, you can disable Data Grid by modifying the CMD to add `--nocluster`
```

File: docs/modules/ROOT/pages/documentation/ecosystem/docker-micro-usage.adoc [hunk 5]
```diff
@@ -0,0 +1,165 @@
...
+
+== Configuration
+
+=== Environment Variables
+
+The following environment variables are available to configure Payara Micro. 
+
+* `MEM_MAX_RAM_PERCENTAGE` - Value for the JVM parameter `-XX:MaxRAMPercentage` that indicate the percentage of memory assigned to the container that can be used by the Java Process. By default this is _70_.
```
Comment (OndroMih) [inline docs/modules/ROOT/pages/documentation/ecosystem/docker-micro-usage.adoc:148]:
```suggestion
* `MEM_MAX_RAM_PERCENTAGE` - Value for the JVM parameter `-XX:MaxRAMPercentage` that indicates the percentage of memory assigned to the container that can be used by the Java Process. By default this is _70_.
```

**Review by OndroMih — CHANGES_REQUESTED** [review]

File: docs/modules/ROOT/pages/documentation/ecosystem/docker-micro-usage.adoc [hunk 6]
```diff
@@ -0,0 +1,165 @@
...
+....
+
+
+=== JDK Version
+
+_Since Payara 5.2020.3_
+
+The Docker Images are available in 2 different flavours. The one is running on JDK 8 and the other has JDK 11 on board.
+
+The command used in the Quick start above, will start the Zulu JDK 8 version. If you need the Zulu JDK 11 version, you need to add the `-jdk11` suffix to the image name.
```
[the remark this answers is not in the document]
Comment (rdebusscher) [reply docs/modules/ROOT/pages/documentation/ecosystem/docker-micro-usage.adoc:28]: It is not a good practice to use the LATEST (or jdk11 tag here) as you do not have control over the moment when a newer version of Payara is used). That is the reason i isn't mentioned.

File: docs/modules/ROOT/pages/documentation/ecosystem/docker-server-usage.adoc [hunk 7]
```diff
@@ -0,0 +1,174 @@
...
+
+The following environment variables are available to configure Payara Server. When edited either in a `Dockerfile` or before the `startInForeground.sh` script is ran, they will change the behaviour of the Payara Server instance.
+
+* `POSTBOOT_COMMANDS` - The name of the file containing post boot commands for the Payara Server instance. This is the file written to in the `generate_deploy_commands.sh` script.
+* `MEM_MAX_RAM_PERCENTAGE` - Value for the JVM parameter `-XX:MaxRAMPercentage` that indicate the percentage of memory assigned to the container that can be used by the Java Process. By default this is _70_.
+* `MEM_XSS` - Defines the value of the Stack size, used in the JVM parameter `-Xss`. The default value is _512k_.
+* `DEPLOY_PROPS` - Specifies a list of properties to be passed with the deploy commands generated in the `generate_deploy_commands.sh` script, For example `'--properties=implicitCdiEnabled=false'`.
+* `PREBOOT_COMMANDS` - The name of the file containing pre boot commands for the Payara Server instance.
+* `ADMIN_PASSWORD` - The password used for accessing the server in combination with user name defined in  `ADMIN-USER`. By default this is `admin`.
```
[the remark this answers is not in the document]
Comment (rdebusscher) [reply docs/modules/ROOT/pages/documentation/ecosystem/docker-server-usage.adoc:107]: ADMIN_PASSWORD is used to define the password for the admin user. See this line in d dockerfile.

````
echo "AS_ADMIN_PASSWORD=\nAS_ADMIN_NEWPASSWORD=${ADMIN_PASSWORD}" > /tmp/password-change-file.txt \
`````

The ADMIN_USER is effectively read only.

File: docs/modules/ROOT/pages/documentation/ecosystem/docker-server-usage.adoc [hunk 8]
```diff
@@ -0,0 +1,174 @@
...
+* `PAYARA_ARGS` - additional arguments to the `start-domain` command that starts the server. Use this only if it’s not enough to specify the configuration using the `POSTBOOT_COMMANDS` or `PREBOOT_COMMANDS` file
+* `JVM_ARGS` - Specifies a list of JVM arguments which will be passed to Payara in the `startInForeground.sh` script.
+* `AS_ADMIN_MASTERPASSWORD` - The master password to pass to Payara Server. This is overriden if one is specified in the `$PASSWORD_FILE`.
```
[the remark this answers is not in the document]
Comment (rdebusscher) [reply docs/modules/ROOT/pages/documentation/ecosystem/docker-server-usage.adoc:110]: correct, was wrong in the old documentation

File: docs/modules/ROOT/pages/documentation/ecosystem/docker-server-usage.adoc (hunk 7 above)
[the remark this answers is not in the document]
Comment (OndroMih) [reply docs/modules/ROOT/pages/documentation/ecosystem/docker-server-usage.adoc]: Yes. This happens at build time when the base image is built. If ADMIN_PASSWORD is set in a derived Dockerfile or when a container is started, it has no effect. That's why it's effectively read only.

If would need to be picked up by one of the init script to have any effect after the base image is built.

File: docs/modules/ROOT/pages/documentation/ecosystem/docker-images.adoc [hunk 9]
```diff
@@ -12,3 +12,10 @@ Docker Images are available here for Payara Micro.
 
 https://hub.docker.com/r/payara/server-web/[Payara Server Web]::
 Docker Images are available here for the web profile of Payara Server.
+
+
+Payara Enterprise customers have access to a private Docker repository containing Docker Images containing the Payara Enterprise Edition. These images are maintained every month. They receive JDK updates, expired Security certificate rates are cleaned and OS updates can be requested.
```
Comment (OndroMih) [inline docs/modules/ROOT/pages/documentation/ecosystem/docker-images.adoc:17]:
```suggestion
TIP: Payara Enterprise customers have access to a private Docker repository containing Docker Images containing the Payara Enterprise Edition. These images are maintained every month. They receive JDK updates, expired Security certificates are removed and OS updates can be requested.
```

**Review by OndroMih — CHANGES_REQUESTED** [review]

**Review by OndroMih — APPROVED** [review]

**Pull request merged**